core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1806,12 +1789,9 @@ class KafkaApis(val requestChannel: RequestChannel,
           createAclsRequest.getErrorResponse(requestThrottleMs,
             new SecurityDisabledException("No Authorizer is configured on the broker.")))
       case Some(auth) =>
-        val errors = mutable.HashMap[Int, Throwable]()
-        val creations = ListBuffer[(Resource, Acl)]()
-        for (i <- 0 until createAclsRequest.aclCreations.size) {
-          val result = toScala(createAclsRequest.aclCreations.get(i).acl.toFilter)
-          result match {
-            case Failure(throwable) => errors.put(i, throwable)
+        val aclCreationResults = createAclsRequest.aclCreations.asScala.map { aclCreation =>
```

Comment:
Yeah, I was wondering about that too. It's an issue that we have no tests to verify this behaviour.